### ISSUE TITLE
A need for downloadable, personal backup copies

### ISSUE THREAD

**Issue by Heikkiss** [thunkable/thunkable-issues#401]
Hi there,

While working somewhat extensively on a Thunkable X project over the past few weeks, I've made numerous copies and shared links to myself to the project so that nothing would happen to it. However, I've felt uneasy about the whole thing the whole time. Reading from the forums how people's projects have suddenly been corrupted (copies included) you can't help but wonder will the same happen to me. After spending so much time and effort, you'd really want to make sure the project is safe. I get it that you want to keep everything online but how about you allowed people to download some kind of a Thunkable project file to the computer. This would let me sleep better... :) After all, you must have such a file on your server already. Overall, what I want to stress is that the lack of hard copies really is an issue from the user perspective.

In other words, currently, as everything is in the same place, on your server, you can see the potential for a catastrophe. Of course, as you develp the platform, some kind of backward compatibility might be an issue, but going forward into the future I just can't see any other reasonable way than allowing people to download Thunkable project files. So, the sooner this would be taken into account, the better. And if you want to limit sharing of these files (for some reason), you could still tie up the file to a certain account.

All in all, this would be a sweet addon!

Thank you,

Heikki

**Comment by josmas** [comment]
Hi @Heikkiss thanks for your request. I am going to copy it and ask the team to weigh in at one of our design meetings. We'll get back to you with whatever is decided.

**Comment by josmas** [comment]
Hi again. We have added this to our TODO list, but unfortunately we do not have the bandwidth to work on it straight away. I'll leave the issue open in any case, and we'll keep you posted when there's any progress. Thanks!

**Comment by Heikkiss** [comment]
Hi @josmas , many thanks for this!

Best, Heikki

**Comment by easaw** [comment]
@Heikkiss for now, have a look at the Screen Saver feature, this allows you to create a copy of your current screen (and import it into another project if required), but the main thing is it's a good snapshot of your current screen and doesn't update.

If you create a good system of versioning (e.g.including date and time), you can make incremental backups of your screens :)

It's working very nicely for me and so far I don't even have very complicated screens!

**Comment by Heikkiss** [comment]
Hi Eric,

Sweet! Tried it out quickly. This is really nice. Not that I wouldn't like hard copies, too, but this gives me more peace of mind--and--it's useful in other ways too. Thank you so much!

Best, Heikki
